numpy/random/mtrand/mtrand.pyx

```diff
@@ -1114,15 +1114,15 @@ cdef class RandomState:
                 # __index__ must return an integer by python rules.
                 pop_size = operator.index(a.item())
             except TypeError:
-                raise ValueError("a must be 1-dimensional or an integer")
-            if pop_size <= 0:
-                raise ValueError("a must be greater than 0")
+                raise ValueError("'a' must be 1-dimensional or an integer")
+            if pop_size <= 0 and np.prod(size) != 0:
+                raise ValueError("'a' must be greater than 0 unless no samples are taken")
         elif a.ndim != 1:
-            raise ValueError("a must be 1-dimensional")
+            raise ValueError("'a' must be 1-dimensional")
```

Comment:
I don't think it is normal to escape argument names in error messages, see, e.g., 

https://github.com/numpy/numpy/blob/master/numpy/random/mtrand/mtrand.pyx#L988

Reply:
My guess is, you can find examples for everything both in numpy and the standard lib. Personally, I think quotes probably make it slightly more discoverable what `a` refers to, so sounds good to me. (e.g. something says `axis is invalid` may not refer to an axis argument, something saying `` `axis` is invalid`` certainly does).

Anyway, I think the PR looks good and you guys can put it in if you like.